OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -2648,19 +2648,52 @@ public static DreamValue NativeProc_sorttext(NativeProc.State state)
         [DreamProc("text2path")]
         [DreamProcParameter("T", Type = DreamValueType.String)]
         public static DreamValue NativeProc_text2path(NativeProc.State state) {
-            if (!state.GetArgument(0, "T").TryGetValueAsString(out var text)) {
+            if (!state.GetArgument(0, "T").TryGetValueAsString(out var text) || text.Length == 0) {
                 return DreamValue.Null;
             }
 
             DreamPath path = new DreamPath(text);
-            if (path.FindElement("proc") != -1 || path.FindElement("verb") != -1)
-                throw new NotImplementedException("text2path() for procs is not implemented");
 
-            if (state.ObjectTree.TryGetTreeEntry(path, out var type)) {
-                return new DreamValue(type);
-            } else {
+            bool isVerb = false;
+            
+            int procElementIndex = path.FindElement("proc");
+            if (procElementIndex == -1) {
+                procElementIndex = path.FindElement("verb");
+                if (procElementIndex != -1)
+                    isVerb = true;
+            }
+
+            bool isProcPath = procElementIndex != -1;
+
+            string? procName = null;
+            if (isProcPath) {
+                // we strip the leading slash here
+                procName = path.FromElements(procElementIndex + 1).ToString()[1..];
```

Comment:
`LastElement` works better here
```suggestion
                procName = path.LastElement;
```